### PR TITLE
Publish security test evidence to assembly repo

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ci/copy-unix-release-artifacts.sh [hunk 1]
```diff
@@ -104,4 +104,8 @@ if [[ "$NAME" == "linux" ]]; then
     cp bazel-bin/docs/sphinx-source-tree.tar.gz $OUTPUT_DIR/split-release/docs/sphinx-source-tree-$RELEASE_TAG.tar.gz
     cp bazel-bin/docs/pdf-fonts-tar.tar.gz $OUTPUT_DIR/split-release/docs/pdf-fonts-$RELEASE_TAG.tar.gz
     cp bazel-bin/docs/non-sphinx-html-docs.tar.gz $OUTPUT_DIR/split-release/docs/non-sphinx-html-docs-$RELEASE_TAG.tar.gz
+
+    bazel build //ledger/test-evidence:generate-security-test-evidence-files
+    cp bazel-bin/ledger/test-evidence/security-test-evidence.csv $OUTPUT_DIR/github/security-test-evidence.csv
```
Comment (soren-da) [inline ci/copy-unix-release-artifacts.sh:109]: consider to prefix the output files with `daml-` such that we can easily differentiate them from the canton ones when processing them in the assembly repo later.

File: ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala [hunk 2]
```diff
@@ -49,54 +33,19 @@ object Main {
...
-
-    val securityTestEntries =
-      collectTestEvidence[SecurityTest, SecurityTestSuite, SecurityTestEntry](
-        testSuites,
-        ledgerApiTests,
-        SecurityTestEntry,
-      )
-
-    val reliabilityTestEntries =
-      collectTestEvidence[ReliabilityTest, ReliabilityTestSuite, ReliabilityTestEntry](
```
Comment (soren-da) [inline ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala:82]: You removed writing the reliability test evidence. Was that intentional? Are there no tests yet annotated with that in the daml repo?

File: ci/copy-unix-release-artifacts.sh (hunk 1 above)
Comment (skisel-da) [reply ci/copy-unix-release-artifacts.sh:109]: sure, will do so

File: ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala (hunk 2 above)
Comment (skisel-da) [reply ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala:82]: Yes, we do not have currently any reliability tests, that is why I removed it not to have empty files

File: ci/build.yml [hunk 3]
```diff
@@ -105,6 +105,11 @@ jobs:
       inputs:
         pathtoPublish: 'bazel-bin/docs/html.tar.gz'
         artifactName: 'Docs bundle'
+    - task: PublishBuildArtifacts@1
+      displayName: 'Publish test evidence'
+      inputs:
+        pathtoPublish: 'bazel-bin/ledger/test-evidence/daml-test-evidence.tar.gz'
+        artifactName: 'Test Evidence'
```
Comment (garyverhaegen-da) [inline ci/build.yml:112]: What are you trying to achieve here? This is not required to publish the artifacts to the assembly repo.

**Review by garyverhaegen-da — COMMENTED** [review]
I don't feel comfortable reviewing the Scala parts of this PR (should it be split?), but I can confirm the changes to `copy-unix-release-artifacts.sh` will result in the test evidence tarball to be pushed to the assembly repo on releases.

File: ci/build.yml (hunk 3 above)
Comment (garyverhaegen-da) [reply ci/build.yml:112]: Actually this will break our ability to make pre-2.2 patch releases, so I'd prefer it be left out.

Comment (skisel-da) [reply ci/build.yml:112]: as agreed with @mziolekda we would like test evidence also being published during every PR build so it can be reviewed

Comment (garyverhaegen-da) [reply ci/build.yml:112]: Are you ever going to actually look at it?

Regardless, as it stands, this is breaking our release process. If you really want to have this, please hide it behind a flag. I recommend removing this from this PR and opening a separate one for that as this one is large enough as it is.

You can add a `condition` to this step based on the value of a variable emitted in a previous Bash step (which you should add) that tests for the existence of the tarball, then uses the `set_var` function from the bash-lib. Ping me on Slack if you need more step-by-step instructions.

Comment (skisel-da) [reply ci/build.yml]: I have removed this change out of this PR

**Comment by skisel-da** [comment]
I moved all Scala changes to #13553, but also based this PR on another PR as Scala one should go first.

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

**Comment by garyverhaegen-da** [comment]
Note: CI will not run on a PR that does not target main. That's fine. It should run again once you change the target to main after merging #13553.

**Pull request merged**